### DISCUSSION TITLE
How to do the versioning of packages in Github actions

### DISCUSSION THREAD

**Discussion by Parwaghu** [community/community#28697] [Actions]
Hello,

I am new to GHA and I am working on a dotnet solution. I want to pack and publish a nuget package in an artifactory instance and have a proper versioning to the packages as 1.0.0, 1.0.1,1.0.2 and so on. 

Currently I am using git tag to pass the latest version number.
On push with git tag like "v1.0.1" from a master branch I want to trigger an dotnet pack command and publish the package to an artifactory instance.

Currently I have this:

on:
  push:
    tags:
    - "v[0-9]+.[0-9]+.[0-9]+"
    ...remaining steps where I use the tag and create and push the newer version of the nuget package
    
With the above set up, this action is getting triggered on each tag push irrespective of the branch. I wan to restrict it to master branch only. So I changed it to the below:

on:
  push:
    branches: [master]
    tags:
    - "v[0-9]+.[0-9]+.[0-9]+"
    ...remaining steps where I use the tag and create and push the newer version of the nuget package
    
   With this change, this workflow is getting triggered twice. Once because of the push to master and second time because of the tag push.
   
How can I achieve it to run only on tag push from master branch?
Is there another way for versioning of the packages?

**Comment by airtower-luna** [comment]
> How can I achieve it to run only on tag push from master branch?

That's not possible: Git has no concept of a tag being on a branch. A tag points at a specific commit, and that's all.

So the first thing you need to decide is what you mean by a tag being on a branch. What you can do then is to write your own check in an additional job, set a job output, and run or don't run the other job(s) [depending on that output](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif).
